packages/flutter_tools/lib/src/device.dart

```diff
@@ -209,7 +209,12 @@ abstract class DeviceManager {
   /// * If the user did not specify a device id and there is more than one
   /// device connected, then filter out unsupported devices and prioritize
   /// ephemeral devices.
-  Future<List<Device>> findTargetDevices(FlutterProject flutterProject) async {
+  Future<List<Device>> findTargetDevices(FlutterProject flutterProject, { Duration timeout }) async {
+    if (timeout != null) {
+      // Reset the cache with the specified timeout.
+      await refreshAllConnectedDevices(timeout: timeout);
```

Comment:
This has the side effect of repopulating the `devices` cache, with the `timeout` duration.  We decided to do this in https://github.com/flutter/flutter/pull/51678#discussion_r387157291 to avoid google3 breakages.